File: algorithms/LengthOfLongestSubstring.java
package algorithms;

import java.util.HashSet;
import java.util.Set;

public class LengthOfLongestSubstring {
    public static int lengthOfLongestSubstring(String s) {
        Set<Character> charSet = new HashSet<>();
        int left = 0, right = 0, maxLength = 0;
        while (right < s.length()) {
            if (!charSet.contains(s.charAt(right))) {
                charSet.add(s.charAt(right));
                right++;
                maxLength = Math.max(maxLength, right - left);
            } else {
                charSet.remove(s.charAt(left));
                left++;
            }
        }
        return maxLength;
    }

    public static void main(String[] args) {
        String input = "abcbbcbb";
        int result = lengthOfLongestSubstring(input);
        System.out.println(
                "Length of longest substring without repeating characters for input \"" + input + "\": " + result);
    }
}

File: algorithmsJS/LengthOfLongestSubstring.js
class LengthOfLongestSubstring {
  static lengthOfLongestSubstring(s) {
    const charSet = new Set();
    let left = 0,
      right = 0,
      maxLength = 0;
    while (right < s.length) {
      if (!charSet.has(s[right])) {
        charSet.add(s[right]);
        right++;
        maxLength = Math.max(maxLength, right - left);
      } else {
        charSet.delete(s[left]);
        left++;
      }
    }
    return maxLength;
  }

  static lengthOfLongestSubstring1(s) {
    return 0;
  }
}

// Example usage:
const input = "abcbbcbb";
const result = LengthOfLongestSubstring.lengthOfLongestSubstring(input);
const result1 = LengthOfLongestSubstring.lengthOfLongestSubstring1(input);
console.log(
  `Length of longest substring without repeating characters for input "${input}": ${result}`
);
console.log(
  `Length of longest substring without repeating characters for input "${input}": ${result1}`
);
